
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.4">
    
    
      
        <title>Mining EMG with R part 2B - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-769c285a91.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-02c2a4388f.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Metagenomics Modules
                </span>
              
                <span class="md-header-nav__parent">
                  Metagenomics mining EMG with R
                </span>
              
            
            Mining EMG with R part 2B
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../timetables/timetable_metagenomics/" title="Timetable" class="md-tabs__link">
          Timetable
        </a>
      
    </li>
  

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-cli/commandline/" title="Metagenomics Modules" class="md-tabs__link md-tabs__link--active">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Timetable
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Timetable
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-4" type="checkbox" id="nav-5-4" checked>
    
    <label class="md-nav__link" for="nav-5-4">
      Metagenomics mining EMG with R
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-4">
        Metagenomics mining EMG with R
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../emgr-part1/" title="Mining EMG with R part 1" class="md-nav__link">
      Mining EMG with R part 1
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../emgr-part2A/" title="Mining EMG with R part 2A" class="md-nav__link">
      Mining EMG with R part 2A
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Mining EMG with R part 2B
      </label>
    
    <a href="./" title="Mining EMG with R part 2B" class="md-nav__link md-nav__link--active">
      Mining EMG with R part 2B
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#3-interproscan-relative-occurence-during-treatment" title="3. InterproScan: relative occurence during treatment." class="md-nav__link">
    3. InterproScan: relative occurence during treatment.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-go-all-category-relative-occurence-during-treatment" title="4. GO (all category): relative occurence during treatment." class="md-nav__link">
    4. GO (all category): relative occurence during treatment.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#50-correspondance-analysisis-of-the-contingency-tables" title="5.0 Correspondance Analysisis of the contingency tables." class="md-nav__link">
    5.0 Correspondance Analysisis of the contingency tables.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#51-perform-a-correspondence-analysis-on-the-metatranscriptomic-data" title="5.1 Perform a Correspondence analysis on the Metatranscriptomic data" class="md-nav__link">
    5.1 Perform a Correspondence analysis on the Metatranscriptomic data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#52-merge-taxonomy-and-biological-processess-and-perform-correspondence-analysis-ca-on-the-merge-set" title="5.2 Merge Taxonomy and Biological processess and perform Correspondence Analysis (CA) on the merge set" class="md-nav__link">
    5.2 Merge Taxonomy and Biological processess and perform Correspondence Analysis (CA) on the merge set
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#53-merge-taxonomy-and-cellular-compartment-and-perform-correspondence-analysis-ca-on-the-merge-set" title="5.3 Merge Taxonomy and Cellular Compartment and perform Correspondence Analysis (CA) on the merge set" class="md-nav__link">
    5.3 Merge Taxonomy and Cellular Compartment and perform Correspondence Analysis (CA) on the merge set
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#54-merge-taxonomy-cellular-compartments-and-molecular-functions-then-perform-correspondence-analysis-ca-on-the-merge-set" title="5.4 Merge Taxonomy, Cellular Compartments and Molecular functions then perform Correspondence Analysis (CA) on the merge set" class="md-nav__link">
    5.4 Merge Taxonomy, Cellular Compartments and Molecular functions then perform Correspondence Analysis (CA) on the merge set
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#60-clustering-of-the-principal-component" title="6.0 Clustering of the Principal Component" class="md-nav__link">
    6.0 Clustering of the Principal Component
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#61-to-find-out-the-relationship-between-annotationsbpcc-species-we-transpose-the-above-table-prior-to-performing-a-ca" title="6.1 To find out the relationship between annotations(BP,CC), species we transpose the above table prior to performing a CA." class="md-nav__link">
    6.1 To find out the relationship between annotations(BP,CC), species we transpose the above table prior to performing a CA.
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#3-interproscan-relative-occurence-during-treatment" title="3. InterproScan: relative occurence during treatment." class="md-nav__link">
    3. InterproScan: relative occurence during treatment.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-go-all-category-relative-occurence-during-treatment" title="4. GO (all category): relative occurence during treatment." class="md-nav__link">
    4. GO (all category): relative occurence during treatment.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#50-correspondance-analysisis-of-the-contingency-tables" title="5.0 Correspondance Analysisis of the contingency tables." class="md-nav__link">
    5.0 Correspondance Analysisis of the contingency tables.
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#51-perform-a-correspondence-analysis-on-the-metatranscriptomic-data" title="5.1 Perform a Correspondence analysis on the Metatranscriptomic data" class="md-nav__link">
    5.1 Perform a Correspondence analysis on the Metatranscriptomic data
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#52-merge-taxonomy-and-biological-processess-and-perform-correspondence-analysis-ca-on-the-merge-set" title="5.2 Merge Taxonomy and Biological processess and perform Correspondence Analysis (CA) on the merge set" class="md-nav__link">
    5.2 Merge Taxonomy and Biological processess and perform Correspondence Analysis (CA) on the merge set
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#53-merge-taxonomy-and-cellular-compartment-and-perform-correspondence-analysis-ca-on-the-merge-set" title="5.3 Merge Taxonomy and Cellular Compartment and perform Correspondence Analysis (CA) on the merge set" class="md-nav__link">
    5.3 Merge Taxonomy and Cellular Compartment and perform Correspondence Analysis (CA) on the merge set
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#54-merge-taxonomy-cellular-compartments-and-molecular-functions-then-perform-correspondence-analysis-ca-on-the-merge-set" title="5.4 Merge Taxonomy, Cellular Compartments and Molecular functions then perform Correspondence Analysis (CA) on the merge set" class="md-nav__link">
    5.4 Merge Taxonomy, Cellular Compartments and Molecular functions then perform Correspondence Analysis (CA) on the merge set
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#60-clustering-of-the-principal-component" title="6.0 Clustering of the Principal Component" class="md-nav__link">
    6.0 Clustering of the Principal Component
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#61-to-find-out-the-relationship-between-annotationsbpcc-species-we-transpose-the-above-table-prior-to-performing-a-ca" title="6.1 To find out the relationship between annotations(BP,CC), species we transpose the above table prior to performing a CA." class="md-nav__link">
    6.1 To find out the relationship between annotations(BP,CC), species we transpose the above table prior to performing a CA.
  </a>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                  <h1>Mining EMG with R part 2B</h1>
                
                <h3 id="3-interproscan-relative-occurence-during-treatment">3. InterproScan: relative occurence during treatment.<a class="headerlink" href="#3-interproscan-relative-occurence-during-treatment" title="Permanent link">&para;</a></h3>
<p>Extract the data from the data structure &ldquo;results&rdquo; created earlier and partition the extracted data per datatype.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ipr <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>IPR_abundances<span class="p">)</span>
ipr<span class="o">&lt;-</span> separateDataType<span class="p">(</span>mydf<span class="o">=</span>ipr<span class="p">,</span> metadata<span class="o">=</span>metadata<span class="p">)</span>
ipr_g <span class="o">&lt;-</span> ipr<span class="o">$</span>genome
ipr_t <span class="o">&lt;-</span> ipr<span class="o">$</span>transcript
ipr_a <span class="o">&lt;-</span> ipr<span class="o">$</span>amplicon
</pre></div>
</td></tr></table>

<p>Calculate the relative occurrence of IPR enrichment per day of treatment for both metagenomic and transcriptomic data types.</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ipr_metagenomics <span class="o">&lt;-</span> ipr_g <span class="o">%&gt;%</span> gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>ipr_g<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
ipr_metatransciptomics <span class="o">&lt;-</span> ipr_t <span class="o">%&gt;%</span> gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">3</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>ipr_t<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
</pre></div>
</td></tr></table>
Merge both metatranscriptomic and metagenomic relative occurrences of IPR per condition to visualise the relative abundance of IPR terms that are dynamic during treatment.
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ipr_genomic_transcriptomics <span class="o">&lt;-</span> <span class="kp">rbind</span><span class="p">(</span>ipr_metagenomics<span class="p">,</span> ipr_metatransciptomics<span class="p">)</span>
ipr_genomic_transcriptomics <span class="o">&lt;-</span> ipr_genomic_transcriptomics <span class="o">%&gt;%</span>
mutate<span class="p">(</span>description<span class="o">=</span><span class="kp">as.character</span><span class="p">(</span>description<span class="p">),</span> count<span class="o">=</span><span class="kp">log2</span><span class="p">(</span>count<span class="m">+1</span><span class="p">))</span>
</pre></div>
</td></tr></table></p>
<p>Bar plot of relative abundance of IPR</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ipr_genomic_transcriptomics <span class="o">%&gt;%</span> arrange<span class="p">(</span>desc<span class="p">(</span>prop<span class="p">))</span> <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span> y<span class="o">=</span>prop<span class="p">,</span>fill<span class="o">=</span>description<span class="p">))</span> <span class="o">+</span>
geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">,</span> position<span class="o">=</span><span class="s">&#39;fill&#39;</span><span class="p">,</span> aes<span class="p">(</span>fill <span class="o">=</span> description<span class="p">))</span> <span class="o">+</span>
facet_wrap<span class="p">(</span><span class="o">~</span>DataType<span class="p">,</span> ncol<span class="o">=</span><span class="m">1</span><span class="p">)</span> <span class="o">+</span>
ggtitle<span class="p">(</span><span class="s">&#39;IPR relative abundance through treatment\n per datatype&#39;</span><span class="p">)</span> <span class="o">+</span>
ylab<span class="p">(</span><span class="s">&#39;proportion abundance&#39;</span><span class="p">)</span> <span class="o">+</span>
theme_light<span class="p">()</span> <span class="o">+</span>
theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">),</span>
axis.text.y<span class="o">=</span>element_blank<span class="p">())</span> <span class="o">+</span>
coord_flip<span class="p">()</span> <span class="o">+</span>
theme<span class="p">(</span>legend.position<span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Heat map of absolute InterproScan hit counts throughout treatment</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ipr_genomic_transcriptomics <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span>y<span class="o">=</span>description<span class="p">))</span><span class="o">+</span> facet_grid<span class="p">(</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
geom_tile<span class="p">(</span>aes<span class="p">(</span>fill<span class="o">=</span>count<span class="p">))</span> <span class="o">+</span> scale_fill_gradient<span class="p">(</span>low<span class="o">=</span><span class="s">&quot;white&quot;</span><span class="p">,</span> high<span class="o">=</span><span class="s">&quot;darkblue&quot;</span><span class="p">)</span> <span class="o">+</span>
xlab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;&quot;</span><span class="p">)</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">),</span>
axis.text.y<span class="o">=</span>element_blank<span class="p">())</span>
</pre></div>
</td></tr></table>

<p><img alt="Heat map of absolute InterproScan hit counts throughout treatment" src="../images/IPR-Abs-heat.png" /></p>
<h3 id="4-go-all-category-relative-occurence-during-treatment">4. GO (all category): relative occurence during treatment.<a class="headerlink" href="#4-go-all-category-relative-occurence-during-treatment" title="Permanent link">&para;</a></h3>
<p>The goal here is to create a composite plot of all GO category occurence during treatments. Give meaningful names to conditions as for the taxonomic case</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12
13
14
15</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>GO <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>GO_abundances<span class="p">)</span>
GO <span class="o">&lt;-</span> GO <span class="o">%&gt;%</span> mutate<span class="p">(</span>description<span class="o">=</span><span class="kp">as.character</span><span class="p">(</span>description<span class="p">),</span> <span class="kt">category</span><span class="o">=</span><span class="kp">as.character</span><span class="p">(</span><span class="kp">category</span><span class="p">))</span>

GO_g <span class="o">&lt;-</span> GO <span class="o">%&gt;%</span>
select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_G|description|category&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metagenomics&quot;</span><span class="p">)</span>
GO_t <span class="o">&lt;-</span> GO <span class="o">%&gt;%</span> select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_T|description|category&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Metatranscriptomics&quot;</span><span class="p">)</span>
GO_a <span class="o">&lt;-</span> GO <span class="o">%&gt;%</span> select<span class="p">(</span>matches<span class="p">(</span><span class="s">&quot;_A|description|category&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>DataType<span class="o">=</span><span class="s">&quot;Amplicon&quot;</span><span class="p">)</span>


<span class="kp">colnames</span><span class="p">(</span>GO_g<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>GO_g<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>GO_t<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>GO_t<span class="p">)</span>
<span class="kp">colnames</span><span class="p">(</span>GO_a<span class="p">)</span> <span class="o">&lt;-</span> renameSample<span class="p">(</span>metadata<span class="o">=</span>metadata<span class="p">,</span> localdf<span class="o">=</span>GO_a<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Transform the data from a wide format to a long format, making use of the tidyr package function gather, and compute the relative frequency of occurrence per days of treatment</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>GO_metagenomics <span class="o">&lt;-</span> GO_g <span class="o">%&gt;%</span> gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">4</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>GO_g<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">,</span> <span class="kp">category</span><span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
GO_metatransciptomics <span class="o">&lt;-</span> GO_t <span class="o">%&gt;%</span> gather<span class="p">(</span>condition<span class="p">,</span> count<span class="p">,</span> <span class="m">4</span><span class="o">:</span><span class="kp">ncol</span><span class="p">(</span>GO_t<span class="p">)</span><span class="m">-1</span><span class="p">)</span> <span class="o">%&gt;%</span>
group_by<span class="p">(</span> condition<span class="p">,</span> <span class="kp">category</span><span class="p">)</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>prop<span class="o">=</span>count<span class="o">/</span><span class="kp">sum</span><span class="p">(</span>count<span class="p">))</span>
</pre></div>
</td></tr></table>

<p>Merge both metatranscriptomic and metagenomic relative occurrences of GO per condition to visualise the relative abundance of all GO category terms that are dynamic during treatment.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>GO_genomic_transcriptomics <span class="o">&lt;-</span> <span class="kp">rbind</span><span class="p">(</span>GO_metagenomics<span class="p">,</span> GO_metatransciptomics<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Barplot of relative abundance of all GO terms</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8
9</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>GO_genomic_transcriptomics <span class="o">%&gt;%</span> arrange<span class="p">(</span>desc<span class="p">(</span>prop<span class="p">))</span> <span class="o">%&gt;%</span>
ggplot<span class="p">(</span> aes<span class="p">(</span>x<span class="o">=</span>condition<span class="p">,</span> y<span class="o">=</span>prop<span class="p">,</span>fill<span class="o">=</span>description<span class="p">))</span> <span class="o">+</span>
geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">,</span> position<span class="o">=</span><span class="s">&#39;fill&#39;</span><span class="p">,</span> aes<span class="p">(</span>fill <span class="o">=</span> description<span class="p">))</span> <span class="o">+</span>
facet_grid<span class="p">(</span><span class="kp">category</span><span class="o">~</span>DataType<span class="p">)</span> <span class="o">+</span>
ggtitle<span class="p">(</span><span class="s">&#39;Gene Ontology relative abundance through days \n of treatment per datatype&#39;</span><span class="p">)</span> <span class="o">+</span>
ylab<span class="p">(</span><span class="s">&#39;proportion abundance&#39;</span><span class="p">)</span> <span class="o">+</span>
theme_light<span class="p">()</span> <span class="o">+</span> theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span> <span class="o">+</span>
coord_flip<span class="p">()</span> <span class="o">+</span>
theme<span class="p">(</span>legend.position<span class="o">=</span><span class="s">&quot;none&quot;</span><span class="p">)</span> <span class="c1">#+ guides(fill=guide_legend(ncol=2))</span>
</pre></div>
</td></tr></table>

<p><img alt="Barplot of relative abundance of all GO terms" src="../images/GO-Bar.png" /></p>
<h3 id="50-correspondance-analysisis-of-the-contingency-tables">5.0 Correspondance Analysisis of the contingency tables.<a class="headerlink" href="#50-correspondance-analysisis-of-the-contingency-tables" title="Permanent link">&para;</a></h3>
<p>The goal of this analysis is to elucidate the relationship between species and GO terms with respect to various treatment conditions (days).
We will make use of the variable created above.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">head</span><span class="p">(</span>genomic_transcriptomics<span class="p">)</span>
taxonomy_long <span class="o">&lt;-</span> genomic_transcriptomics <span class="o">%&gt;%</span> ungroup<span class="p">()</span> <span class="o">%&gt;%</span>
mutate<span class="p">(</span>condition<span class="o">=</span><span class="kp">paste</span><span class="p">(</span>DataType<span class="p">,</span> condition<span class="p">,</span> sep<span class="o">=</span><span class="s">&quot;_&quot;</span><span class="p">))</span> <span class="o">%&gt;%</span>
select<span class="p">(</span>phylum<span class="p">,</span> condition<span class="p">,</span> count<span class="p">)</span>
<span class="kp">head</span><span class="p">(</span>taxonomy_long<span class="p">)</span>
taxonomy_wide <span class="o">&lt;-</span> spread<span class="p">(</span>taxonomy_long<span class="p">,</span> condition<span class="p">,</span> count<span class="p">)</span>
<span class="kp">rownames</span><span class="p">(</span>taxonomy_wide<span class="p">)</span> <span class="o">&lt;-</span> taxonomy_wide<span class="o">$</span>Phylum
taxonomy_wide
taxonomy_wide.df <span class="o">&lt;-</span> select<span class="p">(</span>taxonomy_wide<span class="p">,</span> <span class="o">-</span>phylum<span class="p">)</span> <span class="o">%&gt;%</span>
<span class="kp">as.data.frame</span><span class="p">()</span>
</pre></div>
</td></tr></table>

<p>Transpose the matrix for the purpose of plotting</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>taxonomy<span class="o">&lt;-</span> <span class="kp">t</span><span class="p">(</span>taxonomy_wide.df<span class="p">)</span>
</pre></div>
</td></tr></table>
Define a main plot function to set up a plot for later use</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>setup.plot <span class="o">&lt;-</span> ggplot<span class="p">()</span> <span class="o">+</span> coord_fixed<span class="p">()</span> <span class="o">+</span>
labs<span class="p">(</span>x<span class="o">=</span><span class="s">&quot;Comp1, Axis1&quot;</span><span class="p">,</span> y<span class="o">=</span><span class="s">&quot;Comp2, Axis2&quot;</span><span class="p">)</span> <span class="o">+</span>
geom_hline<span class="p">(</span>yintercept<span class="o">=</span><span class="m">0</span><span class="p">,</span> col<span class="o">=</span><span class="s">&quot;darkgrey&quot;</span><span class="p">)</span> <span class="o">+</span>
geom_vline<span class="p">(</span>xintercept<span class="o">=</span><span class="m">0</span><span class="p">,</span> col<span class="o">=</span><span class="s">&quot;darkgrey&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>
Make the scree plot in a viewport</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>myscree <span class="o">&lt;-</span> <span class="kr">function</span><span class="p">(</span>eigs<span class="p">,</span> x<span class="o">=</span><span class="m">0.8</span><span class="p">,</span> y<span class="o">=</span><span class="m">0.1</span><span class="p">,</span> just<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="s">&quot;right&quot;</span><span class="p">,</span><span class="s">&quot;bottom&quot;</span><span class="p">)){</span>
vp <span class="o">&lt;-</span> viewport<span class="p">(</span>x<span class="o">=</span>x<span class="p">,</span> y<span class="o">=</span>y<span class="p">,</span> width<span class="o">=</span><span class="m">0.2</span><span class="p">,</span> height<span class="o">=</span><span class="m">0.2</span><span class="p">,</span> just<span class="o">=</span>just<span class="p">)</span>
mm <span class="o">&lt;-</span> <span class="kt">data.frame</span><span class="p">(</span>x<span class="o">=</span><span class="kp">factor</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="kp">length</span><span class="p">(</span>eigs<span class="p">)),</span> y<span class="o">=</span>eigs<span class="p">)</span>
sp <span class="o">&lt;-</span> ggplot<span class="p">(</span>mm<span class="p">,</span> aes<span class="p">(</span>x<span class="o">=</span>x<span class="p">,</span> y<span class="o">=</span>y<span class="p">))</span> <span class="o">+</span> geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&quot;identity&quot;</span><span class="p">)</span> <span class="o">+</span>
labs<span class="p">(</span>x <span class="o">=</span> <span class="kc">NULL</span><span class="p">,</span> y <span class="o">=</span> <span class="kc">NULL</span><span class="p">)</span> <span class="o">+</span> theme_light<span class="p">()</span>
<span class="kp">print</span><span class="p">(</span>sp<span class="p">,</span> vp<span class="o">=</span>vp<span class="p">)</span>
<span class="p">}</span>
</pre></div>
</td></tr></table>
Extract various datatype</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>taxonomy.dna <span class="o">&lt;-</span> taxonomy_wide.df <span class="o">%&gt;%</span> select<span class="p">(</span>matches<span class="p">(</span><span class="s">&#39;Metagenomics&#39;</span><span class="p">))</span>
taxonomy.rna <span class="o">&lt;-</span> taxonomy_wide.df <span class="o">%&gt;%</span> select<span class="p">(</span>matches<span class="p">(</span><span class="s">&#39;Metatranscriptomics&#39;</span><span class="p">))</span>
taxonomy.dna<span class="o">&lt;-</span> <span class="kp">t</span><span class="p">(</span>taxonomy.dna<span class="p">)</span>
taxonomy.rna<span class="o">&lt;-</span> <span class="kp">t</span><span class="p">(</span>taxonomy.rna<span class="p">)</span>
</pre></div>
</td></tr></table>
Read in a file describing the data we loaded previously into R</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>data.dna <span class="o">&lt;-</span> metadata<span class="p">[</span>metadata<span class="o">$</span>datatype<span class="o">==</span><span class="s">&quot;Metagenomic&quot;</span> <span class="o">&amp;</span>
<span class="o">!</span><span class="p">(</span>metadata<span class="o">$</span>datatype<span class="o">==</span><span class="s">&quot;Amplicon_DNA&quot;</span><span class="o">|</span> metadata<span class="o">$</span>datatype<span class="o">==</span><span class="s">&quot;Amplicon_RNA&quot;</span><span class="p">),]</span>
data.rna <span class="o">&lt;-</span> metadata<span class="p">[</span>metadata<span class="o">$</span>datatype<span class="o">==</span><span class="s">&quot;Metatranscriptomic&quot;</span> <span class="o">&amp;</span>
<span class="o">!</span><span class="p">(</span>metadata<span class="o">$</span>datatype<span class="o">==</span><span class="s">&quot;Amplicon_DNA&quot;</span><span class="o">|</span> metadata<span class="o">$</span>datatype<span class="o">==</span><span class="s">&quot;Amplicon_RNA&quot;</span><span class="p">),]</span>
</pre></div>
</td></tr></table>

<p>Perform a Correspondance analysis on Metagenomic data</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>taxonomy.coa <span class="o">&lt;-</span> dudi.coa<span class="p">(</span>taxonomy.dna<span class="p">,</span> scannf<span class="o">=</span><span class="bp">F</span><span class="p">,</span> nf<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Plot the correspondence analysis with Ade4 native plotter</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>scatter<span class="p">(</span>taxonomy.coa<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>The plot can be made more visually appealing and easier to interpret with ggplot2, making use of data computed for us by Ade4 (dudi.coa). These are in the object return by the dudi.coa function.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>taxonomy.dna.plot<span class="o">&lt;-</span> setup.plot <span class="o">+</span> geom_point<span class="p">(</span>data<span class="o">=</span><span class="kt">data.frame</span><span class="p">(</span>taxonomy.coa<span class="o">$</span>li<span class="p">,</span> data.dna<span class="p">),</span>
aes<span class="p">(</span>x<span class="o">=</span>Axis1<span class="p">,</span> y<span class="o">=</span>Axis2<span class="p">,</span> col<span class="o">=</span>days<span class="p">,</span> shape<span class="o">=</span>days<span class="p">,</span> size<span class="o">=</span><span class="m">3</span><span class="p">))</span> <span class="o">+</span>
geom_text<span class="p">(</span>data<span class="o">=</span>taxonomy.coa<span class="o">$</span>co<span class="p">,</span>
aes<span class="p">(</span>x<span class="o">=</span>Comp1<span class="p">,</span> y<span class="o">=</span>Comp2<span class="p">,</span> label<span class="o">=</span><span class="kp">rownames</span><span class="p">(</span>taxonomy.coa<span class="o">$</span>co<span class="p">)),</span>
position <span class="o">=</span> <span class="s">&quot;jitter&quot;</span><span class="p">,</span> alpha<span class="o">=</span><span class="m">0.2</span><span class="p">)</span> <span class="o">+</span>
scale_size_area<span class="p">(</span>breaks<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">11</span><span class="p">,</span><span class="m">14</span><span class="p">,</span><span class="m">40</span><span class="p">))</span> <span class="o">+</span>
scale_shape<span class="p">(</span>solid<span class="o">=</span><span class="bp">F</span><span class="p">)</span> <span class="o">+</span>
labs<span class="p">(</span>title<span class="o">=</span><span class="s">&quot;Correspondence Analysis: Metagenomics\nTaxonomy abundance&quot;</span><span class="p">)</span> <span class="o">+</span> theme_light<span class="p">()</span>
taxonomy.dna.plot
myscree<span class="p">(</span>taxonomy.coa<span class="o">$</span>eig <span class="o">/</span> <span class="kp">sum</span><span class="p">(</span>taxonomy.coa<span class="o">$</span>eig<span class="p">))</span>
</pre></div>
</td></tr></table>

<p><img alt="Correspondence Analysis: Metagenomics Taxonomy abundance" src="../images/CA-G.png" /></p>
<h3 id="51-perform-a-correspondence-analysis-on-the-metatranscriptomic-data">5.1 Perform a Correspondence analysis on the Metatranscriptomic data<a class="headerlink" href="#51-perform-a-correspondence-analysis-on-the-metatranscriptomic-data" title="Permanent link">&para;</a></h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>taxonomy.coa <span class="o">&lt;-</span> dudi.coa<span class="p">(</span>taxonomy.rna<span class="p">,</span> scannf<span class="o">=</span><span class="bp">F</span><span class="p">,</span> nf<span class="o">=</span><span class="m">2</span><span class="p">)</span>
taxonomy.rna.plot <span class="o">&lt;-</span> setup.plot <span class="o">+</span> geom_point<span class="p">(</span>data<span class="o">=</span><span class="kt">data.frame</span><span class="p">(</span>taxonomy.coa<span class="o">$</span>li<span class="p">,</span> data.rna<span class="p">),</span>
aes<span class="p">(</span>x<span class="o">=</span>Axis1<span class="p">,</span> y<span class="o">=</span>Axis2<span class="p">,</span> col<span class="o">=</span>days<span class="p">,</span> shape<span class="o">=</span>days<span class="p">,</span> size<span class="o">=</span><span class="m">3</span><span class="p">))</span> <span class="o">+</span>
geom_text<span class="p">(</span>data<span class="o">=</span>taxonomy.coa<span class="o">$</span>co<span class="p">,</span>
aes<span class="p">(</span>x<span class="o">=</span>Comp1<span class="p">,</span> y<span class="o">=</span>Comp2<span class="p">,</span> label<span class="o">=</span><span class="kp">rownames</span><span class="p">(</span>taxonomy.coa<span class="o">$</span>co<span class="p">)),</span>
position <span class="o">=</span> <span class="s">&quot;jitter&quot;</span><span class="p">,</span> alpha<span class="o">=</span><span class="m">0.2</span><span class="p">)</span> <span class="o">+</span>
scale_size_area<span class="p">(</span>breaks<span class="o">=</span><span class="kt">c</span><span class="p">(</span><span class="m">0</span><span class="p">,</span><span class="m">3</span><span class="p">,</span><span class="m">6</span><span class="p">,</span><span class="m">11</span><span class="p">,</span><span class="m">14</span><span class="p">,</span><span class="m">40</span><span class="p">))</span> <span class="o">+</span>
scale_shape<span class="p">(</span>solid<span class="o">=</span><span class="bp">F</span><span class="p">)</span> <span class="o">+</span>
labs<span class="p">(</span>title<span class="o">=</span><span class="s">&quot;Correspondence Analysis: Metatranscriptomic\nTaxonomy abundance&quot;</span><span class="p">)</span> <span class="o">+</span>
theme_light<span class="p">()</span>
taxonomy.rna.plot
myscree<span class="p">(</span>taxonomy.coa<span class="o">$</span>eig <span class="o">/</span> <span class="kp">sum</span><span class="p">(</span>taxonomy.coa<span class="o">$</span>eig<span class="p">))</span>
</pre></div>
</td></tr></table>

<p><img alt="Correspondence Analysis: Metatranscriptomic Taxonomy abundance" src="../images/CA-T.png" /></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
<p>What can you conclude despite the very sparse data?</p>
</div>
<p>Merge taxonomic and GO_slim_abundances for correspondence analysis._
Organise the taxonomy data</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>taxonomy <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>phylum_taxonomy_abundances<span class="p">)</span> <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span>kingdom<span class="p">)</span>
phylum <span class="o">&lt;-</span> taxonomy<span class="o">$</span>phylum
taxonomy <span class="o">&lt;-</span> taxonomy <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span>phylum<span class="p">)</span> <span class="o">%&gt;%</span> <span class="kp">t</span><span class="p">()</span>
<span class="kp">colnames</span><span class="p">(</span>taxonomy<span class="p">)</span> <span class="o">&lt;-</span> phylum
taxonomy <span class="o">&lt;-</span> taxonomy<span class="p">[</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&#39;_A$&#39;</span><span class="p">,</span><span class="kp">rownames</span><span class="p">(</span>taxonomy<span class="p">)),]</span>
</pre></div>
</td></tr></table>

<p>Organise the ontologies data</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ontology.meta <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>GO_slim_abundances<span class="p">)</span> <span class="o">%&gt;%</span>select<span class="p">(</span>GO<span class="p">,</span>description<span class="p">,</span> <span class="kp">category</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Organise the Biological processes data</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>go.bp <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>BP_GO_slim_abundances<span class="p">)</span> <span class="o">%&gt;%</span>select<span class="p">(</span><span class="o">-</span>GO<span class="p">,</span> <span class="o">-</span><span class="kp">category</span><span class="p">)</span>
description.go <span class="o">&lt;-</span> go.bp<span class="o">$</span>description
go.bp <span class="o">&lt;-</span> go.bp <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span>description<span class="p">)</span> <span class="o">%&gt;%</span> <span class="kp">t</span><span class="p">()</span>
<span class="kp">colnames</span><span class="p">(</span>go.bp<span class="p">)</span> <span class="o">&lt;-</span> description.go
</pre></div>
</td></tr></table>

<p>Organise the Molecular function data
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>go.mf <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>MF_GO_slim_abundances<span class="p">)</span> <span class="o">%&gt;%</span>select<span class="p">(</span><span class="o">-</span>GO<span class="p">,</span> <span class="o">-</span><span class="kp">category</span><span class="p">)</span>
description.go <span class="o">&lt;-</span> go.mf<span class="o">$</span>description
go.mf <span class="o">&lt;-</span> go.mf <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span>description<span class="p">)</span> <span class="o">%&gt;%</span> <span class="kp">t</span><span class="p">()</span>
<span class="kp">colnames</span><span class="p">(</span>go.mf<span class="p">)</span> <span class="o">&lt;-</span> description.go
</pre></div>
</td></tr></table></p>
<p>Organise the Cellular compartment data
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>go.cc <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>CC_GO_slim_abundances<span class="p">)</span> <span class="o">%&gt;%</span>select<span class="p">(</span><span class="o">-</span>GO<span class="p">,</span> <span class="o">-</span><span class="kp">category</span><span class="p">)</span>
description.go <span class="o">&lt;-</span> go.cc<span class="o">$</span>description
go.cc <span class="o">&lt;-</span> go.cc <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span>description<span class="p">)</span> <span class="o">%&gt;%</span> <span class="kp">t</span><span class="p">()</span>
<span class="kp">colnames</span><span class="p">(</span>go.cc<span class="p">)</span> <span class="o">&lt;-</span> description.go
</pre></div>
</td></tr></table></p>
<p>Organise the IPR data</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ipr <span class="o">&lt;-</span> tbl_df<span class="p">(</span>results<span class="o">$</span>IPR_abundances<span class="p">)</span> <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span>IPR<span class="p">)</span>
description.ipr <span class="o">&lt;-</span> ipr<span class="o">$</span>description
ipr <span class="o">&lt;-</span> ipr <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span>description<span class="p">)</span> <span class="o">%&gt;%</span> <span class="kp">t</span><span class="p">()</span>
<span class="kp">colnames</span><span class="p">(</span>ipr<span class="p">)</span> <span class="o">&lt;-</span> description.ipr
</pre></div>
</td></tr></table>

<p>Read in the annotation material</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>metainfo <span class="o">&lt;-</span> metadata
</pre></div>
</td></tr></table>

<p>Filter out the amplicons data</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>metainfo <span class="o">&lt;-</span> metainfo <span class="o">%&gt;%</span> filter<span class="p">(</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&#39;_A$&#39;</span><span class="p">,</span> id<span class="p">))</span>
metainfo
</pre></div>
</td></tr></table>

<p>Merge the Taxonomy and Molecular Functions and perform Correspondence Analysis (CA) on the merge set</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mydf <span class="o">&lt;-</span> <span class="kp">cbind</span><span class="p">(</span>taxonomy<span class="p">,</span> go.mf<span class="p">)</span>
mydf <span class="o">&lt;-</span> mydf<span class="p">[</span><span class="kp">rownames</span><span class="p">(</span>mydf<span class="p">)</span> <span class="o">%in%</span> metainfo<span class="o">$</span>id<span class="p">,]</span>
</pre></div>
</td></tr></table>

<p>Call on dudi.coa function from Ade4 package to perform CA</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mydf.coa <span class="o">&lt;-</span> dudi.coa<span class="p">(</span>mydf<span class="p">,</span> scannf<span class="o">=</span><span class="bp">F</span><span class="p">,</span> nf<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Visualize the CA analysis with ggplot2</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What can you conclude from this analysis?</p>
<h3 id="52-merge-taxonomy-and-biological-processess-and-perform-correspondence-analysis-ca-on-the-merge-set">5.2 Merge Taxonomy and Biological processess and perform Correspondence Analysis (CA) on the merge set<a class="headerlink" href="#52-merge-taxonomy-and-biological-processess-and-perform-correspondence-analysis-ca-on-the-merge-set" title="Permanent link">&para;</a></h3>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mydf <span class="o">&lt;-</span> <span class="kp">cbind</span><span class="p">(</span>taxonomy<span class="p">,</span> go.bp<span class="p">)</span>
mydf <span class="o">&lt;-</span> mydf<span class="p">[</span><span class="kp">rownames</span><span class="p">(</span>mydf<span class="p">)</span> <span class="o">%in%</span> metainfo<span class="o">$</span>id<span class="p">,]</span>
mydf.coa <span class="o">&lt;-</span> dudi.coa<span class="p">(</span>mydf<span class="p">,</span> scannf<span class="o">=</span><span class="bp">F</span><span class="p">,</span> nf<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>
</td></tr></table>
Visualize the CA analysis with ggplot2</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What can you conclude from this analysis?</p>
<h3 id="53-merge-taxonomy-and-cellular-compartment-and-perform-correspondence-analysis-ca-on-the-merge-set">5.3 <em>Merge Taxonomy and Cellular Compartment and perform Correspondence Analysis (CA) on the merge set</em><a class="headerlink" href="#53-merge-taxonomy-and-cellular-compartment-and-perform-correspondence-analysis-ca-on-the-merge-set" title="Permanent link">&para;</a></h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mydf <span class="o">&lt;-</span> <span class="kp">cbind</span><span class="p">(</span>taxonomy<span class="p">,</span> go.cc<span class="p">)</span>
mydf <span class="o">&lt;-</span> mydf<span class="p">[</span><span class="kp">rownames</span><span class="p">(</span>mydf<span class="p">)</span> <span class="o">%in%</span> metainfo<span class="o">$</span>id<span class="p">,]</span>
mydf.coa <span class="o">&lt;-</span> dudi.coa<span class="p">(</span>mydf<span class="p">,</span> scannf<span class="o">=</span><span class="bp">F</span><span class="p">,</span> nf<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Visualize the CA analysis with ggplot2</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What can you conclude from this analysis?</p>
<h3 id="54-merge-taxonomy-cellular-compartments-and-molecular-functions-then-perform-correspondence-analysis-ca-on-the-merge-set">5.4 <em>Merge Taxonomy, Cellular Compartments and Molecular functions then perform Correspondence Analysis (CA) on the merge set</em><a class="headerlink" href="#54-merge-taxonomy-cellular-compartments-and-molecular-functions-then-perform-correspondence-analysis-ca-on-the-merge-set" title="Permanent link">&para;</a></h3>
<p>Merging Taxonomy and Cellular compartment and Molecular function</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>mydf <span class="o">&lt;-</span> <span class="kp">cbind</span><span class="p">(</span>taxonomy<span class="p">,</span> go.cc<span class="p">,</span> go.mf<span class="p">)</span>
mydf <span class="o">&lt;-</span> mydf<span class="p">[</span><span class="kp">rownames</span><span class="p">(</span>mydf<span class="p">)</span> <span class="o">%in%</span> metainfo<span class="o">$</span>id<span class="p">,]</span>
mydf.coa <span class="o">&lt;-</span> dudi.coa<span class="p">(</span>mydf<span class="p">,</span> scannf<span class="o">=</span><span class="bp">F</span><span class="p">,</span> nf<span class="o">=</span><span class="m">2</span><span class="p">)</span>
</pre></div>
</td></tr></table>
Visualize the CA analysis with ggplot2</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What can you conclude from this analysis?</p>
<h3 id="60-clustering-of-the-principal-component">6.0 Clustering of the Principal Component<a class="headerlink" href="#60-clustering-of-the-principal-component" title="Permanent link">&para;</a></h3>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>compositedata <span class="o">&lt;-</span> <span class="kp">as.data.frame</span><span class="p">(</span>mydf<span class="p">)</span>
<span class="kp">rownames</span><span class="p">(</span>compositedata<span class="p">)</span> <span class="o">&lt;-</span> <span class="kp">paste</span><span class="p">(</span>metainfo<span class="o">$</span>days<span class="p">,</span> metainfo<span class="o">$</span>genomictype<span class="p">,</span> sep<span class="o">=</span><span class="s">&quot;_&quot;</span> <span class="p">)</span>
</pre></div>
</td></tr></table>
Remove columns that only contain zero as they are not contributing to the creation of the component when using FactoMineR</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>compositedata<span class="o">&lt;-</span> compositedata<span class="p">[,</span> <span class="kp">which</span><span class="p">(</span><span class="o">!</span><span class="kp">apply</span><span class="p">(</span>compositedata<span class="p">,</span><span class="m">2</span><span class="p">,</span>FUN<span class="o">=</span><span class="kr">function</span><span class="p">(</span>x<span class="p">){</span><span class="kp">all</span><span class="p">(</span>x<span class="o">==</span><span class="m">0</span><span class="p">)}))]</span>
</pre></div>
</td></tr></table>
Perform a correspondence analysis using FactoMineR. This automatically generates a perceptual map, showing the relationship between GO terms, species, and days of treatment and various datatypes.
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ca.analysis <span class="o">&lt;-</span> CA<span class="p">(</span>compositedata<span class="p">)</span>
</pre></div>
</td></tr></table>
Perform a hierarchical clustering of the principal component created above to see the relationship between days of experiment and datatypes.
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>clustering<span class="o">&lt;-</span> HCPC<span class="p">(</span>ca.analysis<span class="p">,</span> nb.clust<span class="o">=</span><span class="m">-1</span><span class="p">,</span> order<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</td></tr></table>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>plot<span class="p">(</span>clustering<span class="p">)</span>
</pre></div>
</td></tr></table></p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What do you conclude from this graph?</p>
<h3 id="61-to-find-out-the-relationship-between-annotationsbpcc-species-we-transpose-the-above-table-prior-to-performing-a-ca">6.1 To find out the relationship between annotations(BP,CC), species we transpose the above table prior to performing a CA.<a class="headerlink" href="#61-to-find-out-the-relationship-between-annotationsbpcc-species-we-transpose-the-above-table-prior-to-performing-a-ca" title="Permanent link">&para;</a></h3>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>annoSpecies <span class="o">&lt;-</span> <span class="kp">as.data.frame</span><span class="p">(</span><span class="kp">t</span><span class="p">(</span>compositedata<span class="p">))</span>
<span class="kp">rownames</span><span class="p">(</span>annoSpecies<span class="p">)</span> <span class="o">&lt;-</span> <span class="kp">rownames</span><span class="p">(</span><span class="kp">t</span><span class="p">(</span>compositedata<span class="p">))</span>
</pre></div>
</td></tr></table>

<p>CA</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>anno.ca.analysis <span class="o">&lt;-</span> CA<span class="p">(</span>annoSpecies<span class="p">)</span>
</pre></div>
</td></tr></table>

<p>Perform a hierarchical clustering of the following principal component created above to see the relationship between species and annotation.
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>anno.clustering<span class="o">&lt;-</span> HCPC<span class="p">(</span>anno.ca.analysis<span class="p">,</span> nb.clust<span class="o">=</span><span class="m">-1</span><span class="p">,</span> order<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</td></tr></table></p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>plot<span class="p">(</span>anno.clustering<span class="p">)</span>
</pre></div>
</td></tr></table>

<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What can you conclude from this clustering?</p>
<p><em><strong>APPENDIX</strong></em></p>
<p><em>To reproduce this hands on session, your R sessionInfo() must be identical to the following:</em></p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>sessionInfo<span class="p">()</span>
</pre></div>
</td></tr></table>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../emgr-part2A/" title="Mining EMG with R part 2A" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Mining EMG with R part 2A
              </span>
            </div>
          </a>
        
        
          <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Metagenomics WGA
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-c35428f87f.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>