
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.4">
    
    
      
        <title>Mining EMG with R part 1 - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-769c285a91.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-02c2a4388f.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Metagenomics Modules
                </span>
              
                <span class="md-header-nav__parent">
                  Metagenomics mining EMG with R
                </span>
              
            
            Mining EMG with R part 1
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../timetables/timetable_metagenomics/" title="Timetable" class="md-tabs__link">
          Timetable
        </a>
      
    </li>
  

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-cli/commandline/" title="Metagenomics Modules" class="md-tabs__link md-tabs__link--active">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Timetable
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Timetable
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-4" type="checkbox" id="nav-5-4" checked>
    
    <label class="md-nav__link" for="nav-5-4">
      Metagenomics mining EMG with R
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-4">
        Metagenomics mining EMG with R
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Mining EMG with R part 1
      </label>
    
    <a href="./" title="Mining EMG with R part 1" class="md-nav__link md-nav__link--active">
      Mining EMG with R part 1
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data" title="Data" class="md-nav__link">
    Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-1-exploratory-analysis-of-emg-portal-contingency-tables" title="Part 1: Exploratory analysis of EMG portal contingency tables" class="md-nav__link">
    Part 1: Exploratory analysis of EMG portal contingency tables
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#loading-of-r-packages" title="Loading of R packages" class="md-nav__link">
    Loading of R packages
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#load-emg-input-data-for-analysis" title="Load EMG input data for analysis" class="md-nav__link">
    Load EMG input data for analysis
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#preprocess-the-data" title="Preprocess the data" class="md-nav__link">
    Preprocess the data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-correspondence-analysis-ca" title="Perform Correspondence Analysis (CA)" class="md-nav__link">
    Perform Correspondence Analysis (CA)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#correlation-coefficient" title="Correlation coefficient" class="md-nav__link">
    Correlation coefficient
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hierarchical-clustering-of-principal-components" title="Hierarchical Clustering of principal components" class="md-nav__link">
    Hierarchical Clustering of principal components
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../emgr-part2A/" title="Mining EMG with R part 2A" class="md-nav__link">
      Mining EMG with R part 2A
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../emgr-part2B/" title="Mining EMG with R part 2B" class="md-nav__link">
      Mining EMG with R part 2B
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-fda/fda/" title="Metagenomics Functional Analysis" class="md-nav__link">
      Metagenomics Functional Analysis
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#data" title="Data" class="md-nav__link">
    Data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#part-1-exploratory-analysis-of-emg-portal-contingency-tables" title="Part 1: Exploratory analysis of EMG portal contingency tables" class="md-nav__link">
    Part 1: Exploratory analysis of EMG portal contingency tables
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#loading-of-r-packages" title="Loading of R packages" class="md-nav__link">
    Loading of R packages
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#load-emg-input-data-for-analysis" title="Load EMG input data for analysis" class="md-nav__link">
    Load EMG input data for analysis
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#preprocess-the-data" title="Preprocess the data" class="md-nav__link">
    Preprocess the data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#perform-correspondence-analysis-ca" title="Perform Correspondence Analysis (CA)" class="md-nav__link">
    Perform Correspondence Analysis (CA)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#correlation-coefficient" title="Correlation coefficient" class="md-nav__link">
    Correlation coefficient
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#hierarchical-clustering-of-principal-components" title="Hierarchical Clustering of principal components" class="md-nav__link">
    Hierarchical Clustering of principal components
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="mining-ebi-metagenomics-emg-output-with-r">Mining EBI Metagenomics (EMG) output with R<a class="headerlink" href="#mining-ebi-metagenomics-emg-output-with-r" title="Permanent link">&para;</a></h1>
<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<hr />
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Load requred R packages and input data for analysis from EMG</p>
</li>
<li>
<p>Know how analysis the data with R statistical packages</p>
</li>
</ul>
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<hr />
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>RStudio:  <a href="https://www.rstudio.com/">https://www.rstudio.com/</a></p>
<h3 id="data">Data<a class="headerlink" href="#data" title="Permanent link">&para;</a></h3>
<p>America Gut Project: <a href="https://www.ebi.ac.uk/metagenomics/projects/ERP012803">EMG Project ERP012803</a></p>
<p>Author: B T.F. Alako - Cochrane team: European Nucleotide Archive. </p>
<h3 id="useful-links">Useful Links<a class="headerlink" href="#useful-links" title="Permanent link">&para;</a></h3>
<p>EBI Metagenomics resource (EMG) :   <a href="../www.ebi.ac.uk/metagenomics/">EBI MG Portal</a></p>
<h2 id="part-1-exploratory-analysis-of-emg-portal-contingency-tables">Part 1: Exploratory analysis of EMG portal contingency tables<a class="headerlink" href="#part-1-exploratory-analysis-of-emg-portal-contingency-tables" title="Permanent link">&para;</a></h2>
<hr />
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<p>The American Gut project is the largest crowdsourced citizen science project to date. Fecal, oral, skin, and other body site samples collected from thousands of participants represent the largest human microbiome cohort in existence. Detailed health and lifestyle and diet data associated with each sample is enabling us to deeply examine associations between the human microbiome and factors such as diet (from vegan to near carnivore and everything in between), season, amount of sleep, and disease states such as IBD, diabetes, or autism spectrum disorder-as well as many other factors not listed here. The American Gut project also encompasses the British Gut and Australian Gut projects, widening the cohort beyond North America. As the project continues to grow, we will be able to identify significant associations that would not be possible with smaller, geographically and health/disease status-limited cohorts. <a href="https://www.ebi.ac.uk/metagenomics/projects/ERP012803">EMG Project ERP012803</a></p>
<p>We will explore the data generated by the study above. The data is available at: <a href="https://www.ebi.ac.uk/metagenomics/projects/ERP012803">EMG Project ERP012803</a></p>
<h3 id="loading-of-r-packages">Loading of R packages<a class="headerlink" href="#loading-of-r-packages" title="Permanent link">&para;</a></h3>
<p>First we need to load the required R packages for analysis.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>stringr<span class="p">)))</span>      <span class="c1"># String format</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>ade4<span class="p">)))</span>         <span class="c1"># Multivariate analysis</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>ggplot2<span class="p">)))</span>      <span class="c1"># Fancy plotting</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>grid<span class="p">)))</span>         <span class="c1"># Has the viewport function</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>dplyr<span class="p">)))</span>        <span class="c1"># Data manipulation</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>tidyr<span class="p">)))</span>        <span class="c1"># Data manipulation</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>FactoMineR<span class="p">)))</span>   <span class="c1"># Multivariate analysis</span>
<span class="kp">suppressWarnings</span><span class="p">(</span><span class="kp">suppressMessages</span><span class="p">(</span><span class="kn">require</span><span class="p">(</span>factoextra<span class="p">)))</span>   <span class="c1"># Visualize result of multivariate analysis</span>
</pre></div>
</td></tr></table>

<h3 id="load-emg-input-data-for-analysis">Load EMG input data for analysis<a class="headerlink" href="#load-emg-input-data-for-analysis" title="Permanent link">&para;</a></h3>
<p>We can then load the analysis summary and metadata files for ERP0123803.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>ERP012803 <span class="o">&lt;-</span> <span class="s">&quot;https://www.ebi.ac.uk/metagenomics/projects/ERP012803/download/2.0/export?contentType=text&amp;exportValue=taxonomy_abundances&quot;</span>
ERP012803.meta <span class="o">&lt;-</span> <span class="s">&quot;https://www.ebi.ac.uk/metagenomics/projects/ERP012803/overview/doExport&quot;</span><span class="p">;</span>
gutproject <span class="o">&lt;-</span> tbl_df<span class="p">(</span>read.delim<span class="p">(</span>file<span class="o">=</span>ERP012803<span class="p">))</span>
meta <span class="o">&lt;-</span>  read.csv<span class="p">(</span>file<span class="o">=</span>ERP012803.meta<span class="p">)</span>
</pre></div>
</td></tr></table>

<h3 id="preprocess-the-data">Preprocess the data<a class="headerlink" href="#preprocess-the-data" title="Permanent link">&para;</a></h3>
<ul>
<li>Filter to only retain full taxonomic lineages with a species name</li>
</ul>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject <span class="o">&lt;-</span> gutproject <span class="o">%&gt;%</span> 
              rename<span class="p">(</span>taxonomy<span class="o">=</span>X.SampleID<span class="p">)</span> <span class="o">%&gt;%</span>
                <span class="c1">#mutate(taxonomy=gsub(&quot;.*?s__&quot;,&quot;&quot;,taxonomy)) %&gt;% </span>
                mutate<span class="p">(</span>taxonomy<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;.*?g__&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span>taxonomy<span class="p">))</span> <span class="o">%&gt;%</span>
                mutate<span class="p">(</span>taxonomy<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;;s__&quot;</span><span class="p">,</span><span class="s">&quot;_&quot;</span><span class="p">,</span> taxonomy<span class="p">))</span> <span class="o">%&gt;%</span>
                  filter<span class="p">(</span>taxonomy <span class="o">!=</span><span class="s">&quot;&quot;</span> <span class="o">|</span> taxonomy<span class="o">==</span><span class="s">&#39;_&#39;</span> <span class="p">)</span> <span class="o">%&gt;%</span>
                    filter<span class="p">(</span><span class="o">!</span><span class="kp">grepl</span><span class="p">(</span><span class="s">&#39;_$&#39;</span><span class="p">,</span> taxonomy<span class="p">))</span>
</pre></div>
</td></tr></table>
 - Merge counts of species in the same experimental sample</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject <span class="o">&lt;-</span> gutproject <span class="o">%&gt;%</span> 
                group_by<span class="p">(</span>taxonomy<span class="p">)</span> <span class="o">%&gt;%</span>
                  summarize_each<span class="p">(</span>funs<span class="p">(</span><span class="kp">sum</span><span class="p">))</span> <span class="o">%&gt;%</span>
                    ungroup<span class="p">()</span>
</pre></div>
</td></tr></table>
 - Use the species name as the rowname, this replaces the default numerical name </p>
<div class="admonition note">
<p class="admonition-title">Note</p>
</div>
<p>You can ignore warning: Setting row names on a tibble is deprecated.</p>
<ul>
<li>Transform the wide format data into the long format for the purpose of appending descriptive information</li>
</ul>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.long  <span class="o">&lt;-</span> gutproject <span class="o">%&gt;%</span> 
                      gather<span class="p">(</span>sampleb<span class="p">,</span> freq<span class="p">,</span> ERR1072624<span class="o">:</span>ERR1160857<span class="p">)</span> <span class="o">%&gt;%</span>
                        filter<span class="p">(</span>taxonomy <span class="o">!=</span><span class="s">&quot;Root&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>
 - Retain only the sample description and the Run id from the metadata</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>meta <span class="o">&lt;-</span> meta <span class="o">%&gt;%</span> select<span class="p">(</span>Sample.Description<span class="p">,</span> Run.ID<span class="p">)</span>
</pre></div>
</td></tr></table>
 - What is the composition of the data.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre> 1
 2
 3
 4
 5
 6
 7
 8
 9
10
11
12</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>meta.strip <span class="o">&lt;-</span>  tbl_df<span class="p">(</span>meta<span class="p">)</span> <span class="o">%&gt;%</span> mutate<span class="p">(</span>Sample.Description<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot; &quot;</span><span class="p">,</span><span class="s">&quot;_&quot;</span><span class="p">,</span>Sample.Description<span class="p">),</span>
                         Sample.Description<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;American_Gut_Project_&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span> Sample.Description<span class="p">),</span> 
                         Sample.Description<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;American_&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span> Sample.Description<span class="p">)</span>
                         <span class="p">)</span>
meta.strip <span class="o">&lt;-</span> meta.strip <span class="o">%&gt;%</span> group_by<span class="p">(</span>Sample.Description<span class="p">)</span> <span class="o">%&gt;%</span>
              mutate<span class="p">(</span>count<span class="o">=</span>n<span class="p">())</span> <span class="o">%&gt;%</span> ungroup<span class="p">()</span> <span class="o">%&gt;%</span>
              select<span class="p">(</span>Sample.Description<span class="p">,</span> count<span class="p">)</span>  <span class="o">%&gt;%</span> <span class="kp">unique</span><span class="p">()</span>

meta.strip <span class="o">%&gt;%</span> ggplot<span class="p">(</span>aes<span class="p">(</span>x<span class="o">=</span>Sample.Description<span class="p">,</span> y<span class="o">=</span>count<span class="p">))</span> <span class="o">+</span>  geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&#39;identity&#39;</span><span class="p">)</span> <span class="o">+</span> 
                geom_text<span class="p">(</span>aes<span class="p">(</span>label<span class="o">=</span>count<span class="p">))</span>  <span class="o">+</span> theme_bw<span class="p">()</span> <span class="o">+</span>
                      theme<span class="p">(</span>axis.text.x<span class="o">=</span>element_text<span class="p">(</span>angle<span class="o">=</span><span class="m">45</span><span class="p">,</span> hjust<span class="o">=</span><span class="m">1</span><span class="p">))</span> <span class="o">+</span> 
                      ggtitle<span class="p">(</span><span class="s">&quot;American gut project&quot;</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>What is the data mostly composed of?</p>
<ul>
<li>Append metadata to the gut data</li>
</ul>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.df <span class="o">&lt;-</span> <span class="kp">merge</span><span class="p">(</span>gutproject.long<span class="p">,</span> meta<span class="p">,</span> by.x<span class="o">=</span><span class="s">&#39;sampleb&#39;</span><span class="p">,</span> by.y<span class="o">=</span><span class="s">&#39;Run.ID&#39;</span><span class="p">)</span>
gutproject.df <span class="o">&lt;-</span> tbl_df<span class="p">(</span>gutproject.df<span class="p">)</span>
</pre></div>
</td></tr></table>
 - Clean up the description of sample</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.df <span class="o">&lt;-</span> gutproject.df <span class="o">%&gt;%</span> 
                  mutate<span class="p">(</span>Sample.Description<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot; &quot;</span><span class="p">,</span><span class="s">&quot;_&quot;</span><span class="p">,</span>Sample.Description<span class="p">),</span> 
                   Sample.Description<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;American_Gut_Project_&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span> Sample.Description<span class="p">),</span>
                   Sample.Description<span class="o">=</span><span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;American_&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span> Sample.Description<span class="p">))</span> <span class="o">%&gt;%</span>
                    ungroup<span class="p">()</span> 
</pre></div>
</td></tr></table>
 - Merge taxonomy occurences count per data source</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4
5
6
7
8</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.df <span class="o">&lt;-</span> gutproject.df <span class="o">%&gt;%</span>
                  group_by<span class="p">(</span>Sample.Description<span class="p">,</span>taxonomy<span class="p">)</span> <span class="o">%&gt;%</span> 
                    summarise<span class="p">(</span>cum<span class="o">=</span><span class="kp">sum</span><span class="p">(</span>freq<span class="p">))</span> <span class="o">%&gt;%</span> ungroup<span class="p">()</span>
gutproject.df <span class="o">&lt;-</span> gutproject.df <span class="o">%&gt;%</span>
                  select<span class="p">(</span>taxonomy<span class="p">,</span>Sample.Description<span class="p">,</span>cum<span class="p">)</span>
                  <span class="c1">#filter(Sample.Description !=&quot;Stool_sample&quot;) %&gt;%</span>
                  <span class="c1">#filter(Sample.Description !=&quot;Stool_sample&quot; &amp; Sample.Description !=&quot;Mouth_sample&quot; )</span>
                  <span class="c1">#filter( Sample.Description     !=&quot;Vaginal_mucus_sample&quot; )</span>
</pre></div>
</td></tr></table>
 - Transform the long format to wide format for the purpose of Mutivariate analysis</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.df <span class="o">&lt;-</span> gutproject.df <span class="o">%&gt;%</span>
                    spread<span class="p">(</span>Sample.Description<span class="p">,</span> cum<span class="p">,</span> fill<span class="o">=</span><span class="m">0</span><span class="p">)</span> <span class="o">%&gt;%</span>
                      <span class="kp">as.data.frame</span><span class="p">()</span>
</pre></div>
</td></tr></table>

<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.mat <span class="o">&lt;-</span> gutproject.df <span class="o">%&gt;%</span> select<span class="p">(</span><span class="o">-</span><span class="kt">c</span><span class="p">(</span><span class="m">1</span><span class="p">))</span> <span class="o">%&gt;%</span> <span class="kp">as.matrix</span><span class="p">()</span>
<span class="kp">rownames</span><span class="p">(</span>gutproject.mat<span class="p">)</span> <span class="o">&lt;-</span> <span class="kp">gsub</span><span class="p">(</span><span class="s">&quot;\\[|\\]&quot;</span><span class="p">,</span><span class="s">&quot;&quot;</span><span class="p">,</span> perl<span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> gutproject.df<span class="o">$</span>taxonomy<span class="p">)</span>
</pre></div>
</td></tr></table>
 - Clean up the matrix</p>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>Is there any association between the species and the data source?</p>
<ul>
<li>Perform a chi-square test of independence, this evaluates whether there is a significant dependence between species and biological samples</li>
</ul>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>chisq <span class="o">&lt;-</span> chisq.test<span class="p">(</span>gutproject.mat<span class="p">)</span>
chisq
</pre></div>
</td></tr></table>
The results should look like this.
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>Pearson&#39;s Chi-squared test
data:  gutproject.mat
X-squared = 106730000, df = 8442, p-value &lt; 2.2e-16
</pre></div>
</td></tr></table>
The species and biological samples are statistically significantly associated (p-value=0)</p>
<h3 id="perform-correspondence-analysis-ca">Perform Correspondence Analysis (CA)<a class="headerlink" href="#perform-correspondence-analysis-ca" title="Permanent link">&para;</a></h3>
<p>Correspondence Analysis (CA) is a generalized form of Principal Component Analysis for categorical data. We use CA to reduce the dimension of the data without loosing the most important information. CA is used to graphically visualize rows points and column points in a low dimensional space</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.ca <span class="o">&lt;-</span> CA<span class="p">(</span>gutproject.mat<span class="p">,</span> graph<span class="o">=</span><span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<ul>
<li>Explore the content of the CA output</li>
</ul>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="kp">summary</span><span class="p">(</span>gutproject.ca<span class="p">,</span> nb.dec <span class="o">=</span> <span class="m">2</span><span class="p">,</span> nbelements <span class="o">=</span> <span class="m">4</span><span class="p">,</span> ncp<span class="o">=</span> <span class="m">3</span><span class="p">)</span>
</pre></div>
</td></tr></table>

<p>The result of the function summary() contains the chi-square statistics and 3 tables:</p>
<ul>
<li>Table 1- Eigenvalues, displays the variances and the percentage of vaiances retained by each dimension.</li>
<li>Table 2, Contains the coordinates, the contribution and the cos2 (quality of representation [0-1] of the first 4 active rows variable on the dimension 1, 2 and 3</li>
<li>Table 3: displays the coordinates, the contribution and the cos2 of the first 4 active column variables on the dimension 1, 2 and 3</li>
</ul>
<div class="admonition note">
<p class="admonition-title">Question</p>
</div>
<p>Can you interprete the CA result?</p>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
</div>
<p>correlation coef., chi-square statistic</p>
<h3 id="correlation-coefficient">Correlation coefficient<a class="headerlink" href="#correlation-coefficient" title="Permanent link">&para;</a></h3>
<p>Eigen values is the amount of information retained by each axis.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3
4</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>eig <span class="o">&lt;-</span> get_eigenvalue<span class="p">(</span>gutproject.ca<span class="p">)</span>
trace <span class="o">&lt;-</span> <span class="kp">sum</span><span class="p">(</span>eig<span class="o">$</span>eigenvalue<span class="p">)</span>
cor.coef <span class="o">&lt;-</span> <span class="kp">sqrt</span><span class="p">(</span><span class="kp">trace</span><span class="p">)</span>
cor.coef
</pre></div>
</td></tr></table>

<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>[1] 1.378614
</pre></div>
</td></tr></table>

<p>As a rule of thumb a correlation above 0.2 can be considered important (Bendixen 1995, 567; Healey 2013, 289-290)
The correlation coef. is 1.38 in the American gut project dataset, indicating a strong association between row (species) and column(biological samples variables.). 
A more rigorous approach for examining the association is to use Chi-square statistics. 
The association is highly significant (p-value=0) </p>
<p>Explore the percentages of inertia explained by the CA dimensions, the scree plot</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>fviz_screeplot<span class="p">(</span>gutproject.ca<span class="p">,</span> barfill<span class="o">=</span><span class="s">&#39;white&#39;</span><span class="p">,</span> addlabels<span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span> <span class="o">+</span> theme_bw<span class="p">()</span>
</pre></div>
</td></tr></table>
The point at which the scree plot shows a bend mignt indicate an optimal dimensionality.</p>
<h3 id="hierarchical-clustering-of-principal-components">Hierarchical Clustering of principal components<a class="headerlink" href="#hierarchical-clustering-of-principal-components" title="Permanent link">&para;</a></h3>
<p>Compute hierarchical clustering of species of CA results</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.hcpc <span class="o">&lt;-</span> HCPC<span class="p">(</span>gutproject.ca<span class="p">,</span> graph <span class="o">=</span> <span class="kc">TRUE</span> <span class="p">,</span> nb.clust<span class="o">=</span><span class="m">4</span><span class="p">,</span> order<span class="o">=</span><span class="kc">TRUE</span> <span class="p">)</span>
</pre></div>
</td></tr></table>
 - Visualize the CA results</p>
<p>Symmetric biplot, whereby both rows (blue) and columns (red) are represented in the same space using the principal coordinates. Only the distance between row points or the distance between column points can be reliably interpreted. 
With a symmetric plot, only general statements can be drawn about the pattern. 
The inter-distance between rows and column can not be interpreted.</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>fviz_ca_biplot<span class="p">(</span>gutproject.ca<span class="p">,</span> repel<span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> select.row <span class="o">=</span> <span class="kt">list</span><span class="p">(</span>contrib <span class="o">=</span> <span class="m">95</span><span class="p">),</span> select.col <span class="o">=</span> <span class="kt">list</span><span class="p">(</span>contrib <span class="o">=</span> <span class="m">10</span><span class="p">),</span> geom<span class="o">=</span><span class="s">&quot;text&quot;</span><span class="p">)</span> <span class="o">+</span> theme_minimal<span class="p">()</span>
</pre></div>
</td></tr></table>
 - Remove labels and add cluster centers</p>
<p>A 3D map of the hierarchical clustering of the principal component.</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>plot<span class="p">(</span>gutproject.hcpc<span class="p">,</span> choice <span class="o">=</span><span class="s">&quot;3D.map&quot;</span><span class="p">,</span> draw.tree <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
     ind.names <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> centers.plot <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> 
     title<span class="o">=</span><span class="s">&#39;Hierarchical clustering of the Principal Components&#39;</span><span class="p">)</span>
</pre></div>
</td></tr></table>
 * Hierarchical Clustering of principal component.</p>
<p>Compute hierarchical clustering of biological samples of CA results</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>res.hcpc <span class="o">&lt;-</span> HCPC<span class="p">(</span>gutproject.ca<span class="p">,</span> cluster.CA <span class="o">=</span> <span class="s">&quot;columns&quot;</span><span class="p">,</span> graph <span class="o">=</span> <span class="kc">TRUE</span> <span class="p">,</span> nb.clust<span class="o">=</span><span class="m">4</span><span class="p">,</span> order<span class="o">=</span><span class="kc">TRUE</span> <span class="p">)</span>
</pre></div>
</td></tr></table>
Biological sample cluster in the dataset
 using fviz_cluster and theme_bw()</p>
<p>Transpose the row and column in the gut data and perform a new correspondance analysis</p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>gutproject.ca <span class="o">&lt;-</span> CA<span class="p">(</span><span class="kp">t</span><span class="p">(</span>gutproject.mat<span class="p">),</span> ncp<span class="o">=</span><span class="m">4</span><span class="p">)</span>
</pre></div>
</td></tr></table>
Symmetric biplot </p>
<p><table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>fviz_ca_biplot<span class="p">(</span>gutproject.ca<span class="p">,</span> repel<span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span> select.row <span class="o">=</span> <span class="kt">list</span><span class="p">(</span>contrib <span class="o">=</span> <span class="m">12</span><span class="p">),</span> 
               select.col <span class="o">=</span> <span class="kt">list</span><span class="p">(</span>contrib <span class="o">=</span> <span class="m">95</span><span class="p">),</span> geom<span class="o">=</span><span class="s">&quot;text&quot;</span><span class="p">)</span> <span class="o">+</span>
  theme_minimal<span class="p">()</span>
</pre></div>
</td></tr></table>
 - Remove labels and add cluster centers</p>
<p>3D hierarchical clustering of the principal components.</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>plot<span class="p">(</span>res.hcpc<span class="p">,</span> choice <span class="o">=</span><span class="s">&quot;3D.map&quot;</span><span class="p">,</span> draw.tree <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
     ind.names <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span> centers.plot <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> title<span class="o">=</span><span class="s">&#39;Hierarchical clustering of the Principal Components&#39;</span><span class="p">)</span>
</pre></div>
</td></tr></table>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Metagenomics Taxonomic
              </span>
            </div>
          </a>
        
        
          <a href="../emgr-part2A/" title="Mining EMG with R part 2A" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                Mining EMG with R part 2A
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-c35428f87f.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>