
<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="This contains all material for Markdown version of BTP workshops.">
      
      
      
        <meta name="author" content="Authors">
      
      
        <link rel="shortcut icon" href="../../../assets/images/favicon.png">
      
      <meta name="generator" content="mkdocs-0.16.3, mkdocs-material-1.7.4">
    
    
      
        <title>Metagenomics Functional Analysis - Markdown Manuals for Bioinformatics Training Platform (BTP)</title>
      
    
    
      <script src="../../../assets/javascripts/modernizr-1df76c4e58.js"></script>
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/application-769c285a91.css">
      
        <link rel="stylesheet" href="../../../assets/stylesheets/application-02c2a4388f.palette.css">
      
    
    
      
        
        
        
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
      <link rel="stylesheet" href="../../../css/details.css">
    
    
  </head>
  
  
  
  
    <body data-md-color-primary="indigo" data-md-color-accent="light-blue">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <a href="../../.." title="Markdown Manuals for Bioinformatics Training Platform (BTP)" class="md-icon md-icon--home md-header-nav__button">
          </a>
        
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <span class="md-flex__ellipsis md-header-nav__title">
          
            
              
                <span class="md-header-nav__parent">
                  Metagenomics Modules
                </span>
              
            
            Metagenomics Functional Analysis
          
        </span>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
          
<div class="md-search" data-md-component="search">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" required placeholder="Search" accesskey="s" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset">close</button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result" data-md-lang-search="">
          <div class="md-search-result__meta" data-md-lang-result-none="No matching documents" data-md-lang-result-one="1 matching document" data-md-lang-result-other="# matching documents">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
      
        

  

<nav class="md-tabs md-tabs--active" data-md-component="tabs">
  <div class="md-tabs__inner md-grid">
    <ul class="md-tabs__list">
      
        
  <li class="md-tabs__item">
    
      <a href="../../.." title="Home" class="md-tabs__link">
        Home
      </a>
    
  </li>

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../../timetables/timetable_metagenomics/" title="Timetable" class="md-tabs__link">
          Timetable
        </a>
      
    </li>
  

      
        
      
        
  
  
    <li class="md-tabs__item">
      
        <a href="../../metagenomics-module-cli/commandline/" title="Metagenomics Modules" class="md-tabs__link md-tabs__link--active">
          Metagenomics Modules
        </a>
      
    </li>
  

      
    </ul>
  </div>
</nav>
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    
      <i class="md-icon md-icon--home md-nav__button"></i>
    
    Markdown Manuals for Bioinformatics Training Platform (BTP)
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../../.." title="Home" class="md-nav__link">
      Home
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../trainers/trainers/" title="The Trainers" class="md-nav__link">
      The Trainers
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      Timetable
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        Timetable
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../../timetables/timetable_metagenomics/" title="Metagenomics" class="md-nav__link">
      Metagenomics
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item">
    <a href="../../../preamble/" title="Workshop Information" class="md-nav__link">
      Workshop Information
    </a>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5" checked>
    
    <label class="md-nav__link" for="nav-5">
      Metagenomics Modules
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        Metagenomics Modules
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-cli/commandline/" title="Introduction to Command Line" class="md-nav__link">
      Introduction to Command Line
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-qc/ngs-qc/" title="Metagenomics QC" class="md-nav__link">
      Metagenomics QC
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-tax/tax/" title="Metagenomics Taxonomic" class="md-nav__link">
      Metagenomics Taxonomic
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5-4" type="checkbox" id="nav-5-4">
    
    <label class="md-nav__link" for="nav-5-4">
      Metagenomics mining EMG with R
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="2">
      <label class="md-nav__title" for="nav-5-4">
        Metagenomics mining EMG with R
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-emgr/emgr-part1/" title="Mining EMG with R part 1" class="md-nav__link">
      Mining EMG with R part 1
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-emgr/emgr-part2A/" title="Mining EMG with R part 2A" class="md-nav__link">
      Mining EMG with R part 2A
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-emgr/emgr-part2B/" title="Mining EMG with R part 2B" class="md-nav__link">
      Mining EMG with R part 2B
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-nav__link">
      Metagenomics WGA
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        Metagenomics Functional Analysis
      </label>
    
    <a href="./" title="Metagenomics Functional Analysis" class="md-nav__link md-nav__link--active">
      Metagenomics Functional Analysis
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#what-are-protein-signatures" title="What are protein signatures?" class="md-nav__link">
    What are protein signatures?
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assigning-functional-information-to-metagenomic-sequences" title="Assigning functional information to metagenomic sequences" class="md-nav__link">
    Assigning functional information to metagenomic sequences
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#finding-functional-information-about-samples-on-the-emg-website" title="Finding functional information about samples on the EMG website" class="md-nav__link">
    Finding functional information about samples on the EMG website
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practical" title="Practical" class="md-nav__link">
    Practical
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#browsing-analysed-data-via-the-emg-website" title="Browsing analysed data via the EMG website" class="md-nav__link">
    Browsing analysed data via the EMG website
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emg-comparison-tools" title="EMG comparison tools" class="md-nav__link">
    EMG comparison tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#visualising-taxonomic-data-using-megan" title="Visualising taxonomic data using MEGAN" class="md-nav__link">
    Visualising taxonomic data using MEGAN
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#key-learning-outcomes" title="Key Learning Outcomes" class="md-nav__link">
    Key Learning Outcomes
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#resources-youll-be-using" title="Resources You’ll be Using" class="md-nav__link">
    Resources You’ll be Using
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tools-used" title="Tools Used" class="md-nav__link">
    Tools Used
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#useful-links" title="Useful Links" class="md-nav__link">
    Useful Links
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#introduction" title="Introduction" class="md-nav__link">
    Introduction
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#what-are-protein-signatures" title="What are protein signatures?" class="md-nav__link">
    What are protein signatures?
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#assigning-functional-information-to-metagenomic-sequences" title="Assigning functional information to metagenomic sequences" class="md-nav__link">
    Assigning functional information to metagenomic sequences
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#finding-functional-information-about-samples-on-the-emg-website" title="Finding functional information about samples on the EMG website" class="md-nav__link">
    Finding functional information about samples on the EMG website
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#practical" title="Practical" class="md-nav__link">
    Practical
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#browsing-analysed-data-via-the-emg-website" title="Browsing analysed data via the EMG website" class="md-nav__link">
    Browsing analysed data via the EMG website
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#emg-comparison-tools" title="EMG comparison tools" class="md-nav__link">
    EMG comparison tools
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#visualising-taxonomic-data-using-megan" title="Visualising taxonomic data using MEGAN" class="md-nav__link">
    Visualising taxonomic data using MEGAN
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="emg-functional-data-analysis">EMG Functional Data Analysis<a class="headerlink" href="#emg-functional-data-analysis" title="Permanent link">&para;</a></h1>
<h2 id="key-learning-outcomes">Key Learning Outcomes<a class="headerlink" href="#key-learning-outcomes" title="Permanent link">&para;</a></h2>
<hr />
<p>After completing this module the trainee should be able to:</p>
<ul>
<li>
<p>Understand how EMG provides functional analysis of metagenomic data sets</p>
</li>
<li>
<p>Know where to find and how to interpret analysis results for samples on the EMG website</p>
</li>
<li>
<p>Know how to download the raw sample data and analysis results for use with 3<sup>rd</sup> party visualisation and statistical analysis packages</p>
</li>
</ul>
<h2 id="resources-youll-be-using">Resources You’ll be Using<a class="headerlink" href="#resources-youll-be-using" title="Permanent link">&para;</a></h2>
<hr />
<h3 id="tools-used">Tools Used<a class="headerlink" href="#tools-used" title="Permanent link">&para;</a></h3>
<p>Megan6 :  <a href="http://ab.inf.uni-tuebingen.de/software/megan6/">http://ab.inf.uni-tuebingen.de/software/megan6/</a></p>
<h3 id="useful-links">Useful Links<a class="headerlink" href="#useful-links" title="Permanent link">&para;</a></h3>
<hr />
<p>EBI Metagenomics resource (EMG) : <br />
<a href="../www.ebi.ac.uk/metagenomics/"><a href="http://www.ebi.ac.uk/metagenomics/">www.ebi.ac.uk/metagenomics/</a></a></p>
<h2 id="introduction">Introduction<a class="headerlink" href="#introduction" title="Permanent link">&para;</a></h2>
<hr />
<p>The EBI Metagenomics resource (EMG) provides functional analysis of predicted coding sequences (pCDS) from metagenomic data sets using the InterPro database. InterPro is a sequence analysis resource that predicts protein family membership, along with the presence of important domains and sites. It does this by combining predictive models known as protein signatures from a number of different databases into a single searchable resource. InterPro curators manually integrate the different signatures, providing names and descriptive abstracts and, whenever possible, adding Gene Ontology (GO) terms. Links are also provided to pathway databases, such as KEGG, MetaCyc and Reactome, and to structural resources, such as SCOP, CATH and PDB.</p>
<h3 id="what-are-protein-signatures">What are protein signatures?<a class="headerlink" href="#what-are-protein-signatures" title="Permanent link">&para;</a></h3>
<p>Protein signatures are obtained by modelling the conservation of amino acids at specific positions within a group of related proteins (i.e., a protein family), or within the domains/sites shared by a group of proteins. InterPro’s different member databases use different computational methods to produce protein signatures, and they each have their own particular focus of interest: structural and/or functional domains, protein families, or protein features, such as active sites or binding sites (see Figure 1).</p>
<p><img alt="InterPro member databases grouped by the methods, indicated in white
coloured text, used to construct their signatures. Their focus of
interest is shown in blue
text." src="https://github.com/BPA-CSIRO-Workshops/metagenomics-module-fda/blob/44d812ab576db73abdf0fe3d965d7c7f042d6eca/handout/InterPro.png?raw=true" /></p>
<p><strong>Figure 1. InterPro member databases grouped by the methods, indicated in white coloured text, used to construct their signatures. Their focus of interest is shown in blue text.</strong></p>
<p>Only a subset of the InterPro member databases are used by EMG: Gene3D, TIGRFAMs, Pfam, PRINTS and PROSITE patterns. These databases were selected since, together, they provide both high coverage and offer detailed functional analysis, and have underlying algorithms that can cope with the vast amounts of fragmentary sequence data found in metagenomic datasets.</p>
<h3 id="assigning-functional-information-to-metagenomic-sequences">Assigning functional information to metagenomic sequences<a class="headerlink" href="#assigning-functional-information-to-metagenomic-sequences" title="Permanent link">&para;</a></h3>
<p>Whilst InterPro matches to metagenomic sequence sets are informative in their own right, EMG offers an additional type of analysis in the form of Gene Ontology (GO) terms. The Gene Ontology is made up of 3 structured controlled vocabularies that describe gene products in terms of their associated biological processes, cellular components and molecular functions in a species-independent manner. By using GO terms, scientists working on different species or using different databases can compare datasets, since they have a precisely defined name and meaning
for a particular concept. Terms in the Gene Ontology are ordered into hierarchies, with less specific terms towards the top and more specific terms towards the bottom (see Figure 2).</p>
<p><img alt="GO" src="https://github.com/BPA-CSIRO-Workshops/metagenomics-module-fda/blob/44d812ab576db73abdf0fe3d965d7c7f042d6eca/handout/GO.png?raw=true" />
<strong>Figure 2. An example of GO terms organised into a hierarchy, with terms becoming less specific as the hierarchy is ascended (e.g., alpha-tubulin binding is a type of cytoskeletal binding, which is a type of protein binding). Note that a GO term can have more than one parent term. The Gene Ontology also allows for different types of relationships between terms (such as ‘has part of’ or ‘regulates’). The EMG analysis pipeline only uses the straightforward ‘is a’ relationships.</strong></p>
<p>More information about the GO project can be found
<a href="http://www.geneontology.org/GO.doc.shtml">http://www.geneontology.org/GO.doc.shtml</a></p>
<p>As part of the EMG analysis pipeline, GO terms for molecular function,biological process and cellular component are associated to pCDS in a sample via the InterPro2GO mapping service. This works as follows:
InterPro entries are given GO terms by curators if the terms can be accurately applied to all of the proteins matching that entry. Sequences searched against InterPro are then associated with GO terms by virtue of the entries they match - a protein that matches one InterPro entry with the GO term ‘kinase activity’ and another InterPro entry with the GO term ‘zinc ion binding’ will be annotated with both GO terms.</p>
<h3 id="finding-functional-information-about-samples-on-the-emg-website">Finding functional information about samples on the EMG website<a class="headerlink" href="#finding-functional-information-about-samples-on-the-emg-website" title="Permanent link">&para;</a></h3>
<p>Functional analysis of samples within projects on the EMG website
<a href="../www.ebi.ac.uk/metagenomics/"><a href="http://www.ebi.ac.uk/metagenomics/">www.ebi.ac.uk/metagenomics/</a></a> can be accessed by clicking on the Functional Analysis tab found toward the top of any sample page (see Figure 3 below).</p>
<p><img alt="FA" src="https://github.com/BPA-CSIRO-Workshops/metagenomics-module-fda/blob/44d812ab576db73abdf0fe3d965d7c7f042d6eca/handout/FA.png?raw=true" />
<strong>Figure 3. A Functional analysis tab can be found towards the top of each run page</strong></p>
<p>Clicking on this tab brings up a page displaying sequence features (the number of reads with pCDS, the number of pCDS with InterPro matches, etc), InterPro match information and GO term annotation for the sample, as shown in Figure 4 and 5 below.</p>
<p>InterPro match information for the predicted coding sequences in the sample is shown. The number of InterPro matches are displayed graphically, and as a table that has a text search facility.</p>
<p><img alt="FA_Iterpro" src="https://github.com/BPA-CSIRO-Workshops/metagenomics-module-fda/blob/44d812ab576db73abdf0fe3d965d7c7f042d6eca/handout/FA_Interpro.png?raw=true" />
<strong>Figure 4. Functional analysis of metagenomics data, as shown on the EMG website.</strong></p>
<p><img alt="FA_GO" src="https://github.com/BPA-CSIRO-Workshops/metagenomics-module-fda/blob/44d812ab576db73abdf0fe3d965d7c7f042d6eca/handout/FA_GO.png?raw=true" />
<strong>Figure 5. The GO terms predicted for the sample are displayed. Different graphical representations are available, and can be selected by clicking on the ‘Switch view’ icons.</strong></p>
<p>The Gene Ontology terms displayed graphically on the web site have been ‘slimmed’ with a special GO slim developed for metagenomic data sets. GO slims are cut-down versions of the Gene Ontology, containing a subset of the terms in the whole GO. They give a broad overview of the ontology content without the detail of the specific fine-grained terms.</p>
<p>The full data sets used to generate both the InterPro and GO overview charts, along with a host of additional data (processed reads, pCDS, reads encoding 16S rRNAs, taxonomic analyses, etc) can be downloaded for further analysis by clicking the Download tab, found towards the top of the page (see Figure 6).</p>
<p><img alt="FA_DL" src="https://github.com/BPA-CSIRO-Workshops/metagenomics-module-fda/blob/44d812ab576db73abdf0fe3d965d7c7f042d6eca/handout/FA_DL.png?raw=true" />
<strong>Figure 6. Each sample has a download tab, where the full set of sequences, analyses, summaries and raw data can be downloaded.</strong></p>
<h2 id="practical">Practical<a class="headerlink" href="#practical" title="Permanent link">&para;</a></h2>
<hr />
<h3 id="browsing-analysed-data-via-the-emg-website">Browsing analysed data via the EMG website<a class="headerlink" href="#browsing-analysed-data-via-the-emg-website" title="Permanent link">&para;</a></h3>
<p>For this session, we are going to look at a Metagenome of a microbial consortium obtained from the Tuna oil field in the Gippsland Basin, Australia project. In 2011, fluid samples were collected from the A7A oil well in the Tuna oil field (38°10&rsquo; S, 148°25&rsquo; E) in the Gippsland Basin. A map of the sampling site is shown on the project page: <a href="https://www.ebi.ac.uk/metagenomics/projects/ERP004636">https://www.ebi.ac.uk/metagenomics/projects/ERP004636</a></p>
<p>To get the Tuna project page, follow the above link or open the Metagenomics Portal home page (<strong><a href="https://www.ebi.ac.uk/metagenomics/">https://www.ebi.ac.uk/metagenomics/</a></strong>).</p>
<ol>
<li>enter ‘<strong>ERP004636</strong>’ in the search box on the top right hand side of the page, and follow the link to project. You should now have a Project overview page, which describes the project, submitter contact details, and links to the samples and runs that the project contains. </li>
<li>Click on the Sample Name link (not the Run link) to arrive at the overview page, describing various contextual data, such as the geographic location from which the material was isolated, its collection date, and so on.</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Question 1:</p>
<p>What is the latitude, longitude and depth at which the sample was collected?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>-38.1700 , 148.4200</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 2:</p>
<p>What geographic location does this correspond to?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Australia</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 3:</p>
<p>What environmental ontology (ENVO) identifer and name has the sample material been annotated with?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>ocean water ENVO:00002151</p>
</div>
<p>Now scroll down to the ‘Associated runs’ section of the page. Some samples can have a number of sequencing runs associated with them (for example, corresponding to 16S rRNA amplicon analyses and WGS analyses performed on the same sample). In this case, there is only 1 associated run: <strong>ERR413102</strong>. Click on the Run ID to go to the Run page.</p>
<p>This page has a number of tabs towards the top (Overview, Quality control, Taxonomy analysis, Functional analysis, and Download). </p>
<p>Click on the ‘<strong>Download</strong>’ tab. Right click the file labelled ‘<strong>Predicted CDS (FASTA)</strong>’ link, and save this file to your desktop. Find the file (it should be in your Downloads folder), unzip it and examine it using ‘less’ by typing the following commands in a terminal window:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2
3</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>cd ~/Downloads
gzip –ERR4131021_MERGED_FASTQ_pCDS.faa.gz 
less ERR413102_MERGED_FASTQ_pCDS.faa
</pre></div>
</td></tr></table>

<p>Have a look at one or two of the many sequences it contains. </p>
<p>You can count the total number of sequences in the file by grepping the number of header lines that start with “&gt;”</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>grep -c <span class="s2">&quot;^&gt;&quot;</span> ERR413102_MERGED_FASTQ_pCDS.faa
</pre></div>
</td></tr></table>

<p>In a moment, we will look at the analysis results for this entire batch of sequences, displayed on the EMG website. First, we will attempt to analyse just one of the predicted coding sequences using InterPro (the analysis results on the EMG website summarise these kind of results for hundreds of thousands of sequences).</p>
<p>In a new tab or window, open your web browser and navigate to <a href="http://www.ebi.ac.uk/interpro/">http://www.ebi.ac.uk/interpro/</a>. Copy and paste the following sequence into the text box on the InterPro home page where it says ‘Analyse your sequence’:</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>&gt;HWI-M02024:110:000000000-A8H0K:1:1101:23198:21331-1:N:0:TCAGAGAC_1_267
HLLSYRYAYGKFSSTHEATIGGCFLTKDEELDDHIVKYEIWDTAGKNGTIHLPRCTTSKAYXIQVTWYRNAIAAVVVFDVTSRDSFEK
</pre></div>
</td></tr></table>

<p>Press Search and wait for your results. Your sequence will be run through the InterProScan analysis software, which attempts to match it against all of the different signatures in the InterProScan database.</p>
<div class="admonition note">
<p class="admonition-title">Question 4:</p>
<p>Which protein family does InterProScan predict your sequence belongs to, and what GO terms are predicted to describe its function?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Here is an answer.</p>
</div>
<p>Clicking on the InterPro entry name or IPR accession number will take you to the InterPro entry page for your result, where more information
can be found.</p>
<p>Return to the overview page for <strong>ERR413102</strong>.</p>
<p>Now we are going to look at the functional analysis results for all of the pCDS in the sample. First, we will find the number of sequences that made it through to the functional analysis section of the pipeline.</p>
<p>Click on the Quality control tab. This page displays a series of charts, showing how many sequences passed each quality control step, how many reads were left after clustering, and so on.</p>
<div class="admonition note">
<p class="admonition-title">Question 5:</p>
<p>After all of the quality filtering steps are complete, how many reads were submitted for analysis by the pipeline?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>44,429,715 Reads.</p>
</div>
<p>Next, we will look at the results of the functional predictions for the
pCDS. These can be found under the Functional analysis tab.</p>
<p>Click on the Functional analysis tab and examine the InterPro match section. The top part of this page shows a sequence feature summary, showing the number of reads with predicted coding sequences (pCDS), the number of pCDS with InterPro matches, etc.</p>
<div class="admonition note">
<p class="admonition-title">Question 6:</p>
<p>How many predicted coding sequences (pCDS) are in the run?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>19,838,031 pCDS.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 7:</p>
<p>How many pCDS have InterProScan hits?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>5,754,502 pCDS.</p>
</div>
<p>Scroll down the page to the InterPro match summary section</p>
<div class="admonition note">
<p class="admonition-title">Question 8:</p>
<p>How many different InterPro entries were matched by the pCDS?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>4,207 IPRO hits.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 9:</p>
<p>Why is this figure different to the number of pCDS that have InterProScan hits?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Multiple CDS match an interpro id.</p>
</div>
<p>Next we will examine the GO terms predicted by InterPro for the pCDS in the sample.</p>
<p>Scroll down to the GO term annotation section of the page and examine the 3 bar charts, which correspond to the 3 different components of the Gene Ontology.Selecting the pie chart representation of GO terms makes it easier to visualise the data to find the answer.</p>
<div class="admonition note">
<p class="admonition-title">Question 10:</p>
<p>What are the top 3 biological process terms predicted for the pCDS from the sample?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>biosynthetic process, nitrogen compound metabolism, small molecule metabolic process</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 11:</p>
<p>How many of the WGS reads are predicted to encode 16S rRNAs?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>41,436</p>
</div>
<p>Now we will look at the taxonomic analysis for this run.</p>
<p>Click on the Taxonomy Analysis tab and examine the phylum composition graph and table.</p>
<div class="admonition note">
<p class="admonition-title">Question 12:</p>
<p>What are the top 3 phyla in the run, according to 16S rRNA analysis?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Unassigned 48%, protobacteria 20.81%, Firmicutes 14.721% and Thermotogae 8.19%</p>
</div>
<p>Select the Krona chart view of the data icon. This brings up an interactive chart that can be used to analyse data at different
taxonomic ranks.</p>
<div class="admonition note">
<p class="admonition-title">Question 13:</p>
<p>What is the proportion of Thermoanaerobacteraceae in the population?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Total 161, 55% Clostridia; 41% Firmicutes; 6% of all</p>
</div>
<p>Now we will compare these analyses with those for a WGS practical.</p>
<div class="admonition note">
<p class="admonition-title">Question 14:</p>
<p>What are the differences in Taxonomy and Functional analysis?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>TBD</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 15:</p>
<p>Do the number of pCDS predicted differ? </p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>TBD</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 16:</p>
<p>Do the number of pCDSs with an InterPro match differ?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>TBD</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 17:</p>
<p>Do InterPro entries matched differ?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>TBD</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 18:</p>
<p>Are these figures broadly comparable to those for the previous analysis?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>TBD</p>
</div>
<h3 id="emg-comparison-tools">EMG comparison tools<a class="headerlink" href="#emg-comparison-tools" title="Permanent link">&para;</a></h3>
<p>To look at the differences in slimmed GO terms between the 2 runs. There are two ways to do this. First, you can simply scroll to the bottom of the page and examine the GO term annotation (note - selecting the bar chart representation of GO terms makes it easier to compare different data sets). Alternatively, you can use the comparison tool, which allows direct comparison of runs within a project. The tool can be accessed by clicking on the ‘Comparison Tool’ tab. At present, the tool only compares slimmed GO terms, but will be expanded to cover full GO terms, InterPro annotations, and taxonomic profiles as development of the site continues.</p>
<div class="admonition note">
<p class="admonition-title"><em>Bonus exercise</em></p>
<p>Bonus exercise.</p>
</div>
<ol>
<li>Click on the Comparison tool tab and choose the Ocean Sampling Day (OSD) 2014 project from the sample list and select the OSD80_2014-06- 21_0m_NPL022 - ERR770971 and OSD80_2014-06-21_2m_NPL022 - ERR770970.</li>
</ol>
<div class="admonition note">
<p class="admonition-title">Question 19:</p>
<p>Are there visible differences between the GO terms for these runs. Could there be any biological explanation for this?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Yes, shift in proportion of some functions. yes climate and population density. </p>
</div>
<p>Navigate back and open the taxonomic analysis results tab for each run.</p>
<div class="admonition note">
<p class="admonition-title">Question 20:</p>
<p>How does the taxonomic composition differ between runs? Are any trends in the data consistent with your answer to question 19?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Unassigned dropped from 48% to 13% and polaribacter is Total 931, 80% Flavobacteriaceae; 74% Bacteriodetes; 43% of all</p>
</div>
<h3 id="visualising-taxonomic-data-using-megan">Visualising taxonomic data using MEGAN<a class="headerlink" href="#visualising-taxonomic-data-using-megan" title="Permanent link">&para;</a></h3>
<p>Next, we are going to look at the taxonomic predictions for all of the runs. To do this, we are going to load them into MEGAN. MEGAN is a tool suite that provides metagenomic data analysis and visualization. We are going to use only a small subset of its features, relating to taxonomic comparison. Detailed information on MEGAN and the analyses and visualisations it offers can be found here: 
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>http://ab.inf.uni-tuebingen.de/data/software/megan5/download/manual.pdf
</pre></div>
</td></tr></table></p>
<p>MEGAN can be downloaded from 
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1</pre></div></td><td class="code"><div class="codehilite"><pre><span></span>http://ab.inf.uni-tuebingen.de/software/megan6/
</pre></div>
</td></tr></table>
(there are 2 editions Community and Ultimate. The Community edition is open source and free to download. We will use this version for the tutorial). However, it should already be installed on your Desktop.</p>
<p><strong>Click</strong> on the MEGAN icon on your desktop to load the s/w.</p>
<p>We now need to download the full taxonomic predictions for all of the runs in the Ocean Sampling Day project.</p>
<p>Navigate to the Project: Ocean Sampling Day (OSD) page, using the breadcrumb URL link at the top of the page. Click on the Analysis summary tab, which will take you to a set of tab separated result matrix files, summarising the taxonomic and functional observations for all runs in the project.</p>
<p>Click on the Taxonomic assignments (TSV) link, which will download the corresponding file to your computer. When prompted, choose to save the file in the Downloads folder.</p>
<p>Open the Terminal, navigate to the Downloads directory and take a look at the file you have just downloaded (hit q to exit less):</p>
<table class="codehilitetable"><tr><td class="linenos"><div class="linenodiv"><pre>1
2</pre></div></td><td class="code"><div class="codehilite"><pre><span></span><span class="nb">cd</span> ~/Downloads
less -S ERP009703_taxonomy_abundances_v2.0.tsv
</pre></div>
</td></tr></table>

<p>You will see it is a large matrix file, with abundance counts for each taxonomic lineage for each run in the project. From the MEGAN menu, choose ‘File’ and ‘Import’. Select ‘CSV Import’ and then find the file you have just downloaded and edited.</p>
<p>From the pop up menu, go with the default options in the Format and Separator sections (which should have the ‘Class, Count’ option selected under format, and the separator set as ‘Tab’) and select ‘Taxonomy’ in classification section, then press ‘Apply’.</p>
<p>There are many different visualisations and comparisons that can be performed using MEGAN, so feel free to explore the data using the tool.</p>
<p>Click on the ‘Show chart’ icon and choose ‘Stacked Bar Chart’. This should give you a bar chart, showing the abundance reads for taxonomic lineages across all of the sampling sites.</p>
<div class="admonition note">
<p class="admonition-title">Question 21:</p>
<p>Are the number of classified 16S reads roughly equivalent across all of the different sampling sites?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>No, range from ERR771006 ~10 to ~3,250 ERR771046</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 22:</p>
<p>Which run appears particularly enriched in Polaribacter?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>ERR770970 ~920</p>
</div>
<p>You can change between abundance counts and relative abundance using the Options drop menu and choosing % Percentage Scale or Linear Scale.</p>
<p>Change the chart view to ‘Bubble Chart’. This visualisation can be useful when comparing a large number of samples.</p>
<div class="admonition note">
<p class="admonition-title">Question 23:</p>
<p>Do any samples contain taxa not found in other samples? </p>
</div>
<p>Take a look at the Schlegelella genus.</p>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Yes, schlegelella is only found in ERR770961 (64)</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 24:</p>
<p>Can you discern any patterns in the geographical distribution of certain species (for example, the cluster of samples enriched for the lactobacillus genus compared to other samples)?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>lactobacillus ERR771039-58 Belgian</p>
</div>
<p>We are now going to take a look at which other datasets in EBI Metagenomics that lactobacilli are found in. Point your browser at <a href="https://www.ebi.ac.uk/metagenomics/search/">https://www.ebi.ac.uk/metagenomics/search/</a></p>
<p>This interface allows you to search the project, sample and run related metadata and analysis results for all of the publicly available datasets in the EBI Metagenomics resource.</p>
<p>Click on the ‘Runs’ tab. You should now see a number of run-related metadata search facets on the left hand side of the page, including ‘Organism’. Click on the ‘More&hellip;’ option under Organism, scroll down to ‘Lactobacillus’ in the pop-up window and select the check box next to it. Now click ‘Filter’. The results page should now show all of the runs that have taxonomic matches to lactobacilli in their datasets. The ‘Biome’ facet on the left hand side of the page now shows the number of matching datasets in each biome category (to save space, the 10 biome categories with the most matching datasets are shown by default).</p>
<div class="admonition note">
<p class="admonition-title">Question 25:</p>
<p>Which biome category has the most datasets that contain lactobacilli?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>Host-associated (67,031) human digestive</p>
</div>
<div class="admonition note">
<p class="admonition-title">Question 26:</p>
<p>How well does this correlate with what’s known about these bacteria?</p>
</div>
<div class="admonition success">
<p class="admonition-title">Answer</p>
<p>TBD</p>
</div>
                
                  
                
              
              
                
              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../metagenomics-module-wga/wga/" title="Metagenomics WGA" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                Metagenomics WGA
              </span>
            </div>
          </a>
        
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            &copy; The authors; 2017-
          </div>
        
        powered by
        <a href="http://www.mkdocs.org" title="MkDocs">MkDocs</a>
        and
        <a href="http://squidfunk.github.io/mkdocs-material/" title="Material for MkDocs">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../../assets/javascripts/application-c35428f87f.js"></script>
      
      
      <script>app.initialize({url:{base:"../../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
      
        <script src="../../../js/details.js"></script>
      
    
    
      
    
  </body>
</html>